allennlp/models/bimpm.py

```diff
@@ -0,0 +1,193 @@
+    vocab : ``Vocabulary``
+    text_field_embedder : ``TextFieldEmbedder``
+        Used to embed the ``premise`` and ``hypothesis`` ``TextFields`` we get as input to the
+        model.
+    word_matcher : ``BiMPMMatching``
+        BiMPM matching on the output of word embeddings of premise and hypothesis.
+    encoder1 : ``Seq2SeqEncoder``
+        First encoder layer for the premise and hypothesis
+    matcher_fw1 : ``BiMPMMatching``
+        BiMPM matching for the forward output of first encoder layer
+    matcher_bw1 : ``BiMPMMatching``
+        BiMPM matching for the backward output of first encoder layer
+    encoder2 : ``Seq2SeqEncoder``
+        Second encoder layer for the premise and hypothesis
+    matcher_fw2 : ``BiMPMMatching``
+        BiMPM matching for the forward output of second encoder layer
+    matcher_bw2 : ``BiMPMMatching``
+        BiMPM matching for the backward output of second encoder layer
+    aggregator : ``Seq2VecEncoder``
+        Aggregator of all BiMPM matching vectors
+    classifier_feedforward : ``FeedForward``
+        Fully connected layers for classification.
+    dropout : ``float``
+        Dropout percentage to use.
+    initializer : ``InitializerApplicator``, optional (default=``InitializerApplicator()``)
+        If provided, will be used to initialize the model parameters.
+    regularizer : ``RegularizerApplicator``, optional (default=``None``)
+        If provided, will be used to calculate the regularization penalty during training.
+    """
+    def __init__(self, vocab: Vocabulary,
+                 text_field_embedder: TextFieldEmbedder,
+                 word_matcher: BiMPMMatching,
+                 encoder1: Seq2SeqEncoder,
+                 matcher_fw1: BiMPMMatching,
+                 matcher_bw1: BiMPMMatching,
+                 encoder2: Seq2SeqEncoder,
+                 matcher_fw2: BiMPMMatching,
+                 matcher_bw2: BiMPMMatching,
+                 aggregator: Seq2VecEncoder,
+                 classifier_feedforward: FeedForward,
+                 dropout: float = 0.1,
+                 initializer: InitializerApplicator = InitializerApplicator(),
+                 regularizer: Optional[RegularizerApplicator] = None) -> None:
+        super(BiMPM, self).__init__(vocab, regularizer)
+
+        self.text_field_embedder = text_field_embedder
+
+        self.word_matcher = word_matcher
+
+        self.encoder1 = encoder1
+        self.matcher_fw1 = matcher_fw1
+        self.matcher_bw1 = matcher_bw1
+
+        self.encoder2 = encoder2
+        self.matcher_fw2 = matcher_fw2
+        self.matcher_bw2 = matcher_bw2
+
+        self.aggregator = aggregator
+
+        matching_dim = self.word_matcher.get_output_dim() + \
+                       self.matcher_fw1.get_output_dim() + self.matcher_bw1.get_output_dim() + \
+                       self.matcher_fw2.get_output_dim() + self.matcher_bw2.get_output_dim()
+
+        if matching_dim != self.aggregator.get_input_dim():
```

Comment:
We have a `check_dimensions_match` function that makes these checks a little easier: https://github.com/allenai/allennlp/blob/580dc8b0e2c6491d4d75b54c3b15b34b462e0c67/allennlp/models/esim.py#L82-L83